Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +128,62 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'You cannot create a root node.');
+        Assert::startsWith($path, '/', 'Target path "%s" must be absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentPath = PathHelper::getParentPath($resolvedPath);
+        $parentDocument = $this->getManager()->find(null, $parentPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $parentPath));
+        }
+
+        /** @var PhpcrOdmResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::notNull($resource->getName(), 'The resource needs a name for the creation.');
+
+            $document = $resource->getPayload();
+            $document->setName($resource->getName());
+            if ($document instanceof HierarchyInterface) {
```

Comment:
I still think we should open complete CRUD endpoints on Phpcr and PhpcrOdm here (over resource-rest-bundle) on the complete node/documents to enable all frontend applications like admins, to work on the path based structure. The content-bundle would give a REST endpoint on based on the content's route (and the dynamic routing) to have CRUD on content only. a use case would be the frontend editing. That would be my big picture in 4 lines :-)